File: Ransa.Entidades/JuntaOperativa/Planificacion.cs
using Ransa.Framework;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Ransa.Entidades.JuntaOperativa
{
    public class Planificacion
    {
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBField("IDJTAOPE")]
        public string IDJTAOPE { get; set; }
        /// <summary>
        /// Fecha de Planificación
        /// </summary>
        [DBField("FCINPLN")]
        public decimal FCINPLN { get; set; }
        /// <summary>
        /// Hora Inicio de Planificación
        /// </summary>
        [DBField("HORAINI")]
        public decimal HORAINI { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBField("FCFNPLN")]
        public decimal FCFNPLN { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBField("HORAFIN")]
        public decimal HORAFIN { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBField("CNTTUR3")]
        public decimal CNTTUR3 { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBField("CNTTUR1")]
        public decimal CNTTUR1 { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBField("CNTTUR2")]
        public decimal CNTTUR2 { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBField("AUTH1")]
        public string AUTH1 { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBField("AUTH2")]
        public string AUTH2 { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBField("AUTH3")]
        public string AUTH3 { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBField("AUTH4")]
        public string AUTH4 { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBField("AUTH1OBS")]
        public string AUTH1OBS { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBField("AUTH2OBS")]
        public string AUTH2OBS { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBField("AUTH3OBS")]
        public string AUTH3OBS { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBField("AUTH4OBS")]
        public string AUTH4OBS { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBField("USERCRE")]
        public string USERCRE { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBField("FECHCRE")]
        public decimal FECHCRE { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBField("HORCRE")]
        public decimal HORCRE { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBField("USERUPD")]
        public string USERUPD { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBField("FECHUPD")]
        public decimal FECHUPD { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBField("HORUPD")]
        public decimal HORUPD { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBField("ESTADO")]
        public string ESTADO { get; set; }

    }
    public class PlanificacionQueryInput
    {
        /// <summary>
        /// NRO de Planificacion
        /// </summary>
        [DBParameter("IDJTAOPE", DBDataType.VarChar)]
        public string IDJTAOPE { get; set; }
        /// <summary>
        /// Estado (P=Pendiente, A=Activo)
        /// </summary>
        [DBParameter("ESTADO", DBDataType.VarChar)]
        public string ESTADO { get; set; }
        /// <summary>
        /// Fecha Planificación
        /// </summary>
        [DBParameter("FECHA", DBDataType.Numeric)]
        public decimal FECHA { get; set; }
        /// <summary>
        /// ACCION (T=Todos, F=Fecha)
        /// </summary>
        [DBParameter("ACCION", DBDataType.VarChar)]
        public string ACCION { get; set; }

    }
    public class PlanificacionFULL
    {
        public List<Planificacion> Simple { get; set; }
        public List<PlanificacionDetalle> Detalle { get; set; }
        public PlanificacionFULL()
        {
            Simple = new List<Planificacion>();
            Detalle = new List<PlanificacionDetalle>();
        }
    }
    public class PlanificacionInput
    {
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("IDJTAOPE", DBDataType.VarChar)]
        public string IDJTAOPE { get; set; }
        /// <summary>
        /// Fecha de Planificación
        /// </summary>
        [DBParameter("FCINPLN", DBDataType.Numeric)]
        public decimal FCINPLN { get; set; }
        /// <summary>
        /// Hora Inicio de Planificación
        /// </summary>
        [DBParameter("HORAINI", DBDataType.Numeric)]
        public decimal HORAINI { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("FCFNPLN", DBDataType.Numeric)]
        public decimal FCFNPLN { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("HORAFIN", DBDataType.Numeric)]
        public decimal HORAFIN { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("CNTTUR3", DBDataType.Numeric)]
        public decimal CNTTUR3 { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("CNTTUR1", DBDataType.Numeric)]
        public decimal CNTTUR1 { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("CNTTUR2", DBDataType.Numeric)]
        public decimal CNTTUR2 { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("AUTH1", DBDataType.VarChar)]
        public string AUTH1 { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("AUTH2", DBDataType.VarChar)]
        public string AUTH2 { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("AUTH3", DBDataType.VarChar)]
        public string AUTH3 { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("AUTH4", DBDataType.VarChar)]
        public string AUTH4 { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("USERCRE", DBDataType.VarChar)]
        public string USERCRE { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("FECHCRE", DBDataType.Numeric)]
        public decimal FECHCRE { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("HORCRE", DBDataType.Numeric)]
        public decimal HORCRE { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("USERUPD", DBDataType.VarChar)]
        public string USERUPD { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("FECHUPD", DBDataType.Numeric)]
        public decimal FECHUPD { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("HORUPD", DBDataType.Numeric)]
        public decimal HORUPD { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("ESTADO", DBDataType.VarChar)]
        public string ESTADO { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("ACCION", DBDataType.VarChar)]
        public string ACCION { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("AUTH1OBS", DBDataType.VarChar)]
        public string AUTH1OBS { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("AUTH2OBS", DBDataType.VarChar)]
        public string AUTH2OBS { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("AUTH3OBS", DBDataType.VarChar)]
        public string AUTH3OBS { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("AUTH4OBS", DBDataType.VarChar)]
        public string AUTH4OBS { get; set; }

    }
    public class PlanificacionDetalle
    {
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBField("IDJTAOPE")]
        public string IDJTAOPE { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBField("ORDEN")]
        public decimal ORDEN { get; set; }
        /// <summary>
        /// cod de Nave
        /// </summary>
        [DBField("CODNAVE")]
        public string CODNAVE { get; set; }
        /// <summary>
        /// codNave de servicio
        /// </summary>
        [DBField("NOMNAVE")]
        public string NOMNAVE { get; set; }
        /// <summary>
        /// codNave de servicio
        /// </summary>
        [DBField("CONTENE")]
        public string CONTENE { get; set; }
        /// <summary>
        /// codNave de servicio
        /// </summary>
        [DBField("OPEPORTU")]
        public string OPEPORTU { get; set; }
        /// <summary>
        /// codNave de servicio
        /// </summary>
        [DBField("CLASE")]
        public string CLASE { get; set; }
        /// <summary>
        /// codNave de servicio
        /// </summary>
        [DBField("TAMANIO")]
        public decimal TAMANIO { get; set; }
        /// <summary>
        /// codNave de servicio
        /// </summary>
        [DBField("PESOMAN")]
        public decimal PESOMAN { get; set; }
        /// <summary>
        /// codNave de servicio
        /// </summary>
        [DBField("TIPOCONT")]
        public string TIPOCONT { get; set; }
        /// <summary>
        /// codNave de servicio
        /// </summary>
        [DBField("REFRIGER")]
        public string REFRIGER { get; set; }
        /// <summary>
        /// codNave de servicio
        /// </summary>
        [DBField("TIPOPLAN")]
        public string TIPOPLAN { get; set; }
        /// <summary>
        /// codNave de servicio
        /// </summary>
        [DBField("ESTADO")]
        public string ESTADO { get; set; }

    }
    public class PlanificacionDetalleInput
    {
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("IDJTAOPE", DBDataType.VarChar)]
        public string IDJTAOPE { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("ORDEN", DBDataType.Numeric)]
        public decimal ORDEN { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("CODNAVE", DBDataType.VarChar)]
        public string CODNAVE { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("CONTENE", DBDataType.VarChar)]
        public string CONTENE { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("CLASE", DBDataType.VarChar)]
        public string CLASE { get; set; }
        /// <summary>
        /// Operador Portuario
        /// </summary>
        [DBParameter("OPEPORTU", DBDataType.VarChar)]
        public string OPEPORTU { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("TAMANIO", DBDataType.Numeric)]
        public decimal TAMANIO { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("PESOMAN", DBDataType.Numeric)]
        public decimal PESOMAN { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("TIPOCONT", DBDataType.VarChar)]
        public string TIPOCONT { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("REFRIGER", DBDataType.VarChar)]
        public string REFRIGER { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("FCHFNDSC", DBDataType.Numeric)]
        public Nullable<decimal> FCHFNDSC { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("HORFNDSC", DBDataType.Numeric)]
        public Nullable<decimal> HORFNDSC { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("TIPOPLAN", DBDataType.VarChar)]
        public string TIPOPLAN { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("FCHCUTOFF", DBDataType.Numeric)]
        public Nullable<decimal> FCHCUTOFF { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("HORCUTOFF", DBDataType.Numeric)]
        public Nullable<decimal> HORCUTOFF { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("FCHCTOFFR", DBDataType.Numeric)]
        public Nullable<decimal> FCHCTOFFR { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("HORCTOFFR", DBDataType.Numeric)]
        public Nullable<decimal> HORCTOFFR { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("ESTADO", DBDataType.VarChar)]
        public string ESTADO { get; set; }
        /// <summary>
        /// Orden de servicio
        /// </summary>
        [DBParameter("ACCION", DBDataType.VarChar)]
        public string ACCION { get; set; }

    }
    public class PlanificacionDetalleQueryInput
    {
        /// <summary>
        /// NRO de Planificacion
        /// </summary>
        [DBParameter("IDJTAOPE", DBDataType.VarChar)]
        public string IDJTAOPE { get; set; }
        /// <summary>
        /// Estado (P=Pendiente, A=Activo)
        /// </summary>
        [DBParameter("ESTADO", DBDataType.VarChar)]
        public string ESTADO { get; set; }
        /// <summary>
        /// ACCION (T=Todos, F=Fecha)
        /// </summary>
        [DBParameter("ACCION", DBDataType.VarChar)]
        public string ACCION { get; set; }

    }
}

File: JuntaOperaciones/Scripts/planificacion.js
var ListAutorizaciones = [];


$(document).ready(function () {
    $('#CboPlanificacion').dropdown();
    $('#Clave').passwordify({ maxLength: 20 });
    
    $('#btnTransporte').attr('disabled', true);
    $('#btnMaquinaria').attr('disabled', true);
    $('#btnOperaciones').attr('disabled', true);
    $('#btnServicios').attr('disabled', true);

    if (AutoR == 1) {
        $('#btnTransporte').attr('disabled', false);
    } else if (AutoR == 2) {
        $('#btnMaquinaria').attr('disabled', false);
    } else if (AutoR == 3) {
        $('#btnOperaciones').attr('disabled', false);
    } else if (AutoR == 4) {
        $('#btnServicios').attr('disabled', false);
    }

    $('#CboPlanificacion').on('change', function () {
        var id = $('#CboPlanificacion').dropdown().value();
        $('#btnTransporte').attr('disabled', true);
        $('#btnMaquinaria').attr('disabled', true);
        $('#btnOperaciones').attr('disabled', true);
        $('#btnServicios').attr('disabled', true);

        $('#userTransporte').removeClass('fa-check');
        $('#userTransporte').addClass('fa-user-clock');

        $('#userMaquinaria').removeClass('fa-check');
        $('#userMaquinaria').addClass('fa-user-clock');
;
        $('#userOperaciones').removeClass('fa-check');
        $('#userOperaciones').addClass('fa-user-clock');

        $('#userServicios').removeClass('fa-check');
        $('#userServicios').addClass('fa-user-clock');

        if (AutoR == 1) {
            $('#btnTransporte').attr('disabled', false);
        } else if (AutoR == 2) {
            $('#btnMaquinaria').attr('disabled', false);
        } else if (AutoR == 3) {
            $('#btnOperaciones').attr('disabled', false);
        } else if (AutoR == 4) {
            $('#btnServicios').attr('disabled', false);
        }
        TraerPlanificacion(id);
    });

    $('#Observacion').on('keyup', function () {
        $("#nCaracteres").text("Caracteres restantes: " + (250 - $(this).val().length));
    });

    $('#btnAprobar').on('click', function () {


        var antiForgeryToken = $("input[name='__RequestVerificationToken']").val();
        var param = {
            ACCION: $('#ACCION').val(),
            IDJTAOPE: $('#CboPlanificacion').dropdown().value(),
            AUTH1: $('#Clave').data('val'),
            AUTH1OBS: $('#Observacion').val()
        }

        function exito(rpta) {
            if (rpta.mensaje == "OK") {
                $('#Clave').val('');
                $('#Clave').data('val','');
                $('#AutorizacionModal').modal('hide');
                HelperFN.stickyShow('Aprobado con exito.', 'success');
                //window.location.href = './planificacion/index';
            } else {
                $('#Clave').val('');
                $('#Clave').data('val', '');
                HelperFN.stickyShow(rpta.mensaje, 'error');
            }
            
        }

        function error(rpta) {

        }


        HelperFN.AjaxJson('POST', './Planificacion/AccionesPlanificacion', param, true, exito, error, antiForgeryToken);
    });

    $('#AutorizacionModal').on('shown.bs.modal', function (event) {

        if ($('#CboPlanificacion').dropdown().value() != "") {
            var button = $(event.relatedTarget) // Button that triggered the modal
            var recipient = button.data('whatever') // Extract info from data-* attributes
            $('#Clave').val('');
            $('#Observacion').val('');
            $('#Clave').data('val', '');
            if (recipient == "A1") {
                $('#ACCION').val(1);
            } else if (recipient == "A2") {
                $('#ACCION').val(2);
            } else if (recipient == "A3") {
                $('#ACCION').val(3);
            } else if (recipient == "A4") {
                $('#ACCION').val(4);
            }
        } else {
            $('#AutorizacionModal').modal('hide');
            HelperFN.stickyShow('Debe seleccionar una planificación.', 'notice');
        }

    });
    $('#AcuerdosModal').on('show.bs.modal', function (event) {
        console.log(ListAutorizaciones,',')


        if ($('#CboPlanificacion').dropdown().value() != "") {

            $('#Observacion1').val(ListAutorizaciones[0]['AUTH1OBS']);
            $('#Observacion2').text(ListAutorizaciones[0].AUTH2OBS);
            $('#Observacion3').val(ListAutorizaciones[0].AUTH3OBS);
            $('#Observacion4').val(ListAutorizaciones[0].AUTH4OBS);

        } else {
            $('#AutorizacionModal').modal('hide');
            HelperFN.stickyShow('Debe seleccionar una planificación.', 'notice');
        }

    });
});


function TraerPlanificacion(ID) {

    var antiForgeryToken = $("input[name='__RequestVerificationToken']").val();
    var param = {
        ACCION: 'U',
        ESTADO: '',
        FECHA: 0,
        IDJTAOPE: ID
    }

    function exito(rpta) {
        ListAutorizaciones = rpta['Simple'];
        ListDet = rpta['Detalle']


        if (rpta['Simple'].length != 0) {
            LlenarDetalle(ListDet)
            if (rpta['Simple'].AUTH1 != null) {
                $('#userTransporte').removeClass('fa-user-clock');
                $('#userTransporte').addClass('fa-check');
                $('#btnTransporte').attr('disabled', true);
            }
            if (rpta['Simple'].AUTH2 != null) {
                $('#userMaquinaria').removeClass('fa-user-clock');
                $('#userMaquinaria').addClass('fa-check');
                $('#btnMaquinaria').attr('disabled', true);
            }
            if (rpta['Simple'].AUTH3 != null) {
                $('#userOperaciones').removeClass('fa-user-clock');
                $('#userOperaciones').addClass('fa-check');
                $('#btnOperaciones').attr('disabled', true);
            }
            if (rpta['Simple'].AUTH4 != null) {
                $('#userServicios').removeClass('fa-user-clock');
                $('#userServicios').addClass('fa-check');
                $('#btnServicios').attr('disabled', true);
            }

            $("#fechaInicio").text(HelperFN.FormatoFecha(rpta['Simple'][0].FCINPLN));
            $("#horaInicio").text(HelperFN.FormatoHora(rpta['Simple'][0].HORAINI));
            $("#fechaFin").text(HelperFN.FormatoFecha(rpta['Simple'][0].FCFNPLN));
            $("#horaFin").text(HelperFN.FormatoHora(rpta['Simple'][0].HORAFIN));
            $("#turno3").text(rpta['Simple'][0].CNTTUR3);
            $("#turno1").text(rpta['Simple'][0].CNTTUR1);
            $("#turno2").text(rpta['Simple'][0].CNTTUR2);
            var Total = rpta['Simple'][0].CNTTUR3 + rpta.Simple[0].CNTTUR1 + rpta['Simple'][0].CNTTUR2
            $("#contenedores").text(Total);
        } else {
         
            $('#btnTransporte').attr('disabled', true);
            $('#btnMaquinaria').attr('disabled', true);
            $('#btnOperaciones').attr('disabled', true);
            $('#btnServicios').attr('disabled', true);

            $('#userTransporte').removeClass('fa-check');
            $('#userTransporte').addClass('fa-user-clock');

            $('#userMaquinaria').removeClass('fa-check');
            $('#userMaquinaria').addClass('fa-user-clock');
            ;
            $('#userOperaciones').removeClass('fa-check');
            $('#userOperaciones').addClass('fa-user-clock');

            $('#userServicios').removeClass('fa-check');
            $('#userServicios').addClass('fa-user-clock');

            if (AutoR == 1) {
                $('#btnTransporte').attr('disabled', false);
            } else if (AutoR == 2) {
                $('#btnMaquinaria').attr('disabled', false);
            } else if (AutoR == 3) {
                $('#btnOperaciones').attr('disabled', false);
            } else if (AutoR == 4) {
                $('#btnServicios').attr('disabled', false);
            }
            $("#fechaInicio").text(0);
            $("#horaInicio").text(0);
            $("#fechaFin").text(0);
            $("#horaFin").text(0);
            $("#turno3").text(0);
            $("#turno1").text(0);
            $("#turno2").text(0);
            $("#contenedores").text(0);
        }
        

    }

    function error(rpta) {

    }


    HelperFN.AjaxJson('POST', './Planificacion/GetPlanificaciones', param, true, exito, error, antiForgeryToken);

}


function LlenarDetalle(datos) {
    if ($.fn.DataTable.isDataTable('#tblDetalle')) {
        $('#tblDetalle').DataTable().destroy();
        $('#tblDetalle thead').empty();
        $('#tblDetalle tbody').empty();
    }

    var Cabecera = '<tr>' +
        '<th scope="col" class="">Orden Serv.</th>' +
        '<th scope="col" class="">Nave</th>' +
        '<th scope="col" class="">Contenedor</th>' +
        '<th scope="col" class="">Ope. Portuario</th>' +
        '<th scope="col" class="">Clase</th>' +
        '<th scope="col" class="">Tamaño</th>' +
        '<th scope="col" class="">Peso Br. Manifestado</th>' +
        '<th scope="col" class="">TIPO</th>' +
        '<th scope="col" class="">Refrigerado</th>' +
        '<th scope="col" class="">Impo / Expo</th>' +
        '</tr>';
    $('#tblDetalle thead').empty();
    $('#tblDetalle thead').append(Cabecera);

    var table = $('#tblDetalle').DataTable({
        destroy: true,
        responsive: true,
        data: datos,
        columnDefs: [

            { responsivePriority: 1, targets: 0 },
            { responsivePriority: 2, targets: -1 },
            {

                "targets": -1,
                "visible": true,
                "searchable": false,
                "orderable": false,
            },
        ],
        columns: [
            { data: 'ORDEN' },
            { data: 'NOMNAVE' },
            { data: 'CONTENE' },
            { data: 'OPEPORTU' },
            { data: 'CLASE' },
            { data: 'TAMANIO' },
            {
                "mData": null,
                "bSortable": true,
                "mRender": function (o) {
                    if (o.PESOMAN == 0) {
                        return 'No Disp.';
                    }
                    else {
                        return o.PESOMAN;
                    }
                }
            },            
            {
                "mData": 'TIPOCONT',
                "bSortable": true,
                "mRender": function (o) {
                    var Valor = '';
                    if (o == 'G') {
                        Valor = 'General'
                    }
                    else if (o == 'C') {
                        Valor = 'Carga Peligrosa'
                    }
                    else if (o == 'R') {
                        Valor = 'Refrigerado'
                    }
                    else if (o == 'Q') {
                        Valor = 'IQBF'
                    }
                    else if (o == '1') {
                        Valor = 'IQBF/Carga Peligrosa'
                    }
                    else if (o == '2') {
                        Valor = 'Refrigerado/Carga Peligrosa'
                    }
                    return Valor;

                }
            },
            {
                "mData": 'REFRIGER',
                "bSortable": true,
                "mRender": function (o) {
                    var Valor = '';
                    if (o == 'S') {
                        Valor = 'SI'
                    }
                    else {
                        Valor = 'NO'
                    }

                    return Valor;
                }
            },
            {
                "mData": 'TIPOPLAN',
                "bSortable": true,
                "mRender": function (o) {
                    var Valor = '';
                    if (o == 'I') {
                        Valor = 'Importación'
                    }
                    else {
                        Valor = 'Exportación'
                    }

                    return Valor;
                }
            },
           
        ],
        language: {
            "decimal": "",
            "emptyTable": "No hay información",
            "info": "Mostrando _START_ a _END_ de _TOTAL_ Entradas",
            "infoEmpty": "Mostrando 0 to 0 of 0 Entradas",
            "infoFiltered": "(Filtrado de _MAX_ total entradas)",
            "infoPostFix": "",
            "thousands": ",",
            "lengthMenu": "Mostrar _MENU_ Entradas",
            "loadingRecords": "Cargando...",
            "processing": "Procesando...",
            "search": "Buscar:",
            "zeroRecords": "Sin resultados encontrados",
            "paginate": {
                "first": "|<<",
                "last": ">>|",
                "next": ">",
                "previous": "<"
            }
        }
    });

}